src/condor_starter.V6.1/starter.cpp

```diff
@@ -4066,9 +4087,14 @@ Starter::removeTempExecuteDir(int& exit_code)
 			int closed = dprintf_close_logs_in_directory(execute_dir.GetFullPath(), true);
 			if (closed) { dprintf(D_FULLDEBUG, "Closed %d logs in %s\n", closed, execute_dir.GetFullPath()); }
 
-			dprintf(D_FULLDEBUG, "Removing %s\n", execute_dir.GetFullPath());
-			if (!execute_dir.Remove_Current_File()) {
-				has_failed = true;
+			if (it->second == "/" && move_to) {
+				dprintf(D_STATUS, "Renaming %s to %s instead of deleting it\n", execute_dir.GetFullPath(), move_to);
+				rename(execute_dir.GetFullPath(), move_to);
```

Comment:
Should probably detect failure here like with the remove case.

Reply:
returning has_failed here would set the starter exit code to cleanup failure, which in turn could mark the slot as broken or the startd would just delete the sandbox.